hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -363,6 +371,35 @@ public RecordsRunningHashLeaf runningHashLeaf() {
 		return metadata.getUniqueTreasuryOwnershipAssociations();
 	}
 
+	private void updateLinks() {
+		var accounts = accounts();
+		var tokenRels = tokenAssociations();
+
+		for (var accountId : accounts.keySet()) {
+			var merkleAccount = accounts.getForModify(accountId);
+
+			EntityNumPair prevAssociationKey = MISSING_NUM_PAIR;
+			for (var tokenId : merkleAccount.tokens().asTokenIds()) {
+				var associationKey = EntityNumPair.fromLongs(accountId.longValue(), tokenId.getTokenNum());
+				var association = tokenRels.getForModify(associationKey);
+				association.setKey(associationKey);
+				association.setNextKey(prevAssociationKey);
+
+				if (prevAssociationKey != MISSING_NUM_PAIR) {
+					var prevAssociation = tokenRels.getForModify(prevAssociationKey);
+					prevAssociation.setPrevKey(associationKey);
+					tokenRels.put(prevAssociationKey, prevAssociation);
```

Comment:
Since we got `prevAssociation` via `getForModify()`, there's no need to then call `put()` after modifying it.